application/src/main/java/org/togetherjava/tjbot/features/roleapplication/ApplicationCreateCommand.java

```diff
@@ -0,0 +1,297 @@
+    private static final String VALUE_DELIMITER = "_";
+    private static final int ARG_COUNT = 3;
+
+    private final ApplicationApplyHandler applicationApplyHandler;
+    private final RoleApplicationSystemConfig roleApplicationSystemConfig;
+
+    /**
+     * Constructs a new {@link ApplicationCreateCommand} with the specified configuration.
+     * <p>
+     * This command is designed to generate an application form for members to apply for roles.
+     *
+     * @param config the configuration containing the settings for the application form
+     */
+    public ApplicationCreateCommand(Config config) {
+        super("application-form", "Generates an application form for members to apply for roles.",
+                CommandVisibility.GUILD);
+
+        this.roleApplicationSystemConfig = config.getRoleApplicationSystemConfig();
+
+        generateRoleOptions(getData());
+        applicationApplyHandler = new ApplicationApplyHandler(roleApplicationSystemConfig);
+    }
+
+    /**
+     * Populates a {@link SlashCommandData} object with the proper arguments.
+     *
+     * @param data the object to populate
+     */
+    private void generateRoleOptions(SlashCommandData data) {
+        IntStream.range(1, OPTIONAL_ROLES_AMOUNT + 1).forEach(index -> {
+            data.addOption(OptionType.STRING, generateOptionId("title", index),
+                    "The title of the role");
+            data.addOption(OptionType.STRING, generateOptionId("description", index),
+                    "The description of the role");
+            data.addOption(OptionType.STRING, generateOptionId("emoji", index),
+                    "The emoji of the role");
+        });
+    }
+
+    private static String generateOptionId(String name, int id) {
+        return "%s%s%d".formatted(name, VALUE_DELIMITER, id);
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandInteractionEvent event) {
+        if (!handleHasPermissions(event)) {
+            return;
+        }
+
+        final List<OptionMapping> optionMappings = event.getInteraction().getOptions();
+        if (optionMappings.isEmpty()) {
+            event.reply("You have to select at least one role.").setEphemeral(true).queue();
+            return;
+        }
+
+        long incorrectArgsCount = getIncorrectRoleArgsCount(optionMappings);
+        if (incorrectArgsCount > 0) {
+            event.reply("Missing information for %d roles.".formatted(incorrectArgsCount))
+                .setEphemeral(true)
+                .queue();
+            return;
+        }
+
+        sendMenu(event);
+    }
+
+    @Override
+    public void onStringSelectSelection(StringSelectInteractionEvent event, List<String> args) {
+        SelectOption selectOption = event.getSelectedOptions().getFirst();
+
+        if (selectOption == null) {
+            return;
+        }
+
+        OffsetDateTime timeSentCache = applicationApplyHandler.getApplicationSubmitCooldown()
```

Comment:
Let's extract this out into it's own method for checking if the user is able to invoke the command. Keep's this method a bit more tidy and the code more readable.